lib/cfg/cfg_Node.mli

```diff
@@ -0,0 +1,54 @@
+(** The module provides the abstraction of a CFG node. *)
+module Node : sig
+  type 'a t [@@deriving show, sexp]
+
+  (** Checks if two nodes and their children are the same recursively.
+
+      Notes: Should always work for the structure generated by {{! Cfg.Converter}Converter}.
+      Severely limited, only checks isomorphism for two children for each node,
+      also doesn't return true for every isomophic graphs that have two
+      children for each node, only for some *)
+  val compare : 'a t -> 'a t -> bool
+
+  (** [length node] computes the number of nodes in the whole nodes' tree *)
+  val length : 'a t -> int
+
+  (** [make data succ prec] creates a new unique node using [data] as its content,
+    [succ] the list of successors and [prec] the list of precedessors identifiers. *)
+  val make : 'a -> 'a t list -> int list -> 'a t
+
+  (** [make id data succ prec] creates a new node using [id] as its identifiers, [data] as its content,
+    [succ] the list of successors and [prec] the list of precedessors identifiers.
+    
+    *Warning* if the same id as another node is chosen, other methods might return wrong results.
+   *)
+  val make_with_id : int -> 'a -> 'a t list -> int list -> 'a t
```

Comment:
Why is the possibility of setting a custom id exposed?

Reply:
Because to create a looping structure manually is otherwise impossible unless you provide the AST and convert, but in the tests we need the manual approach.